pkg/pgmodel/end_to_end_tests/migrate_test.go

```diff
@@ -40,3 +44,151 @@ func TestMigrateTwice(t *testing.T) {
 		performMigrate(t, connectURL)
 	})
 }
+
+func verifyLogs(t testing.TB, db *pgxpool.Pool, expected []string) {
+	rows, err := db.Query(context.Background(), "SELECT msg FROM log ORDER BY id")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	found := make([]string, 0)
+	for rows.Next() {
+		var value string
+		err = rows.Scan(&value)
+		if err != nil {
+			t.Fatal(err)
+		}
+		found = append(found, value)
+	}
+	if !reflect.DeepEqual(expected, found) {
+		t.Errorf("wrong values in DB\nexpected:\n\t%v\ngot:\n\t%v", expected, found)
+	}
+}
+
+func TestMigrationLib(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	testhelpers.WithDB(t, *testDatabase, testhelpers.NoSuperuser, func(db *pgxpool.Pool, t testing.TB, connectURL string) {
+		testTOC := map[string][]string{
+			"idempotent": {
+				"2-toc-run_first.sql",
+				"1-toc-run_second.sql",
+			},
+		}
+
+		expected := []string{
+			"setup",
+			"idempotent 1",
+			"idempotent 2",
+			"migration 0.2.0",
+			"idempotent 1",
+			"idempotent 2",
+			"idempotent 1",
+			"idempotent 2",
+			"migration 0.9.0",
+			"migration 0.10.0=1",
+			"migration 0.10.0=2",
+			"idempotent 1",
+			"idempotent 2",
+			"idempotent 1",
+			"idempotent 2",
+			"idempotent 1",
+			"idempotent 2",
+			"migration 0.10.1=1",
+			"idempotent 1",
+			"idempotent 2",
+			"migration 0.10.1=2",
+			"idempotent 1",
+			"idempotent 2",
+			"migration 0.10.2-beta=1",
+			"idempotent 1",
+			"idempotent 2",
+		}
```

Comment:
I like this test, but keeping all the values in one static slice makes in incredibly difficult to determine what the individual migrations are supposed to do. I recommend  starting with the log values for the initial migration and `append`ing the later values as we need them, e.g.
```go
expected := []string{
	"setup",
	"idempotent 1",
	"idempotent 2",
}

mig := pgmodel.NewMigrator(db, test_migrations.MigrationFiles, testTOC)

err := mig.Migrate(semver.MustParse("0.1.1"))
if err != nil {
	t.Fatal(err)
}

verifyLogs(t, db, expected)

...

expected = append(expected,
	"migration 0.2.0",
	"idempotent 1",
	"idempotent 2",
)

err = mig.Migrate(semver.MustParse("0.2.0"))
if err != nil {
	t.Fatal(err)
}
verifyLogs(t, db, expected)
...
```
That'll make it very simple to determine what the effect of each migration should be.